### PR TITLE
all, lib/logger: Refactor SetDebug calls

### PR DESCRIPTION
**Pull request by lukaslihotzki**
There are a lot of similar SetDebug calls which can be integrated into NewFacility. This pull request should not change behavior. I tested it with some, but not all facilities.

### CODE REVIEW

File: lib/fs/debug.go [hunk 1]
```diff
@@ -19,11 +16,7 @@ var (
 
 func init() {
 	logger.DefaultLogger.NewFacility("walkfs", "Filesystem access while walking")
-	switch {
-	case strings.Contains(os.Getenv("STTRACE"), "walkfs") || os.Getenv("STTRACE") == "all":
-		l.SetDebug("walkfs", true)
-		fallthrough
-	case strings.Contains(os.Getenv("STTRACE"), "fs"):
+	if logger.DefaultLogger.ShouldDebug("walkfs") {
```
Comment (AudriusButkevicius) [inline lib/fs/debug.go]: Just setdebug(... ShouldDebug())

Comment (AudriusButkevicius) [reply lib/fs/debug.go]: Also, do we need this? Because "fs" is in "walkfs"

File: lib/sync/debug.go [hunk 2]
```diff
@@ -23,13 +22,11 @@ var (
 	// We make an exception in this package and have an actual "if debug { ...
 	// }" variable, as it may be rather performance critical and does
 	// nonstandard things (from a debug logging PoV).
-	debug       = strings.Contains(os.Getenv("STTRACE"), "sync") || os.Getenv("STTRACE") == "all"
+	debug       = logger.DefaultLogger.IsTraced("sync")
```
Comment (AudriusButkevicius) [inline lib/sync/debug.go]: l.ShouldDebug()?

File: lib/api/debug.go [hunk 3]
```diff
@@ -24,5 +21,7 @@ func shouldDebugHTTP() bool {
 func init() {
 	// The debug facility was originally named "http", changed in:
 	// https://github.com/syncthing/syncthing/pull/5548
-	l.SetDebug("api", strings.Contains(os.Getenv("STTRACE"), "api") || strings.Contains(os.Getenv("STTRACE"), "http") || os.Getenv("STTRACE") == "all")
+	if l.IsTraced("http") {
```
Comment (AudriusButkevicius) [inline lib/api/debug.go]: I think we can remove this these days.

File: lib/logger/logger.go [hunk 4]
```diff
@@ -50,6 +50,7 @@ type Logger interface {
 	Warnf(format string, vals ...interface{})
 	ShouldDebug(facility string) bool
 	SetDebug(facility string, enabled bool)
+	IsTraced(facility string) bool
```
Comment (AudriusButkevicius) [inline lib/logger/logger.go]: Based on the comments I think we could get rid of this

File: lib/api/debug.go (hunk 3 above)
Comment (imsodin) [reply lib/api/debug.go]: I would keep it: The change isn't that old and it doesn't really hurt.

File: lib/fs/debug.go (hunk 1 above)
Comment (lukaslihotzki) [reply lib/fs/debug.go]: `l.SetDebug("fs", logger.DefaultLogger.ShouldDebug("walkfs"))` would disable `fs` tracing if `walkfs` is not traced.

Relying on "fs" being in "walkfs" feels like a hack. Also, [postSystemDebug](https://github.com/syncthing/syncthing/blob/master/lib/api/api.go#L615) can set just `walkfs` without setting `fs`. Sadly, this code is run just once before postSystemDebug.

**Comment by lukaslihotzki** [comment]
Somehow the last commit failed CI. I don't know why. Everything seems fine locally.

**Comment by imsodin** [comment]
That's the coverage runner that somehow flakes out on TestCSRFRequired every now and then, nothing to worry about.

File: lib/api/debug.go (hunk 3 above)
Comment (AudriusButkevicius) [reply lib/api/debug.go]: But removing it does not cause damage

**Comment by calmh** [comment]
This is a rare beauty, having a -10x line ratio. Good stuff.

Screen Shot 2019-10-03 at 20 09 04

**Pull request merged**